config.go

```diff
@@ -42,3 +121,91 @@ func (c *configuration) GetTagNames() []string {
 	}
 	return names
 }
+
+func (c *configuration) validate() error {
+	switch {
+	case c.Consul == nil:
+		return fmt.Errorf("No Consul configuration")
+	case c.Consul.URL == "":
+		return fmt.Errorf("No Consul URL specified")
+	case c.Cloud == nil:
+		return fmt.Errorf("No cloud configuration")
+	case c.Cloud.Project == "":
+		return fmt.Errorf("No cloud project specified")
+	case c.Cloud.Network == "":
+		return fmt.Errorf("No cloud network specified")
+	case c.Cloud.Zone == "":
+		return fmt.Errorf("No cloud zone specified")
+	case c.Cloud.URLMap == "":
+		return fmt.Errorf("No cloud URL map specified")
```

Comment:
Same here, I would suggest to add an explicit `default` in a switch.

Reply:
Same comment, let's discuss the thingy in the conversation above.